doc/modules/ensemble.rst

```diff
@@ -1158,3 +1157,87 @@ The following example shows how to fit the VotingRegressor::
 .. topic:: Examples:
 
   * :ref:`sphx_glr_auto_examples_ensemble_plot_voting_regressor.py`
+
+.. _stacking:
+
+Stacked generalization
+======================
+
+Stacked generalization is a method for combining estimators to reduce their
+biases [W1992]. More precisely, the predictions of each individual estimator
+are stacked together and used as input to a final estimator to compute the
+final prediction. This final estimator is trained through cross-validation.
+
+The :class:`StackingClassifier` and :class:`StackingRegressor` provide such
+strategies which can be applied to classification and regression problems.
+
+The `estimators` parameter corresponds to the list of the estimators which
+are stacked together. It should be given as a list of names and estimators::
+
+  >>> from sklearn.linear_model import RidgeCV, LassoCV
+  >>> from sklearn.svm import SVR
+  >>> estimators = [('ridge', RidgeCV()),
+  ...               ('lasso', LassoCV(random_state=42)),
+  ...               ('svr', SVR(C=1, gamma=1e-6, kernel='rbf'))]
+
+The `final_estimator` will use the predictions of the `estimators` as input. It
+needs to be a classifier or a regressor when using :class:`StackingClassifier`
+or :class:`StackingRegressor`, respectively::
+
+  >>> from sklearn.ensemble import GradientBoostingRegressor
+  >>> from sklearn.ensemble import StackingRegressor
+  >>> reg = StackingRegressor(
+  ...     estimators=estimators,
+  ...     final_estimator=GradientBoostingRegressor(random_state=42))
+
+To train the `estimators` and `final_estimator`, the `fit` method needs
+to be called on the training data::
+
+  >>> from sklearn.datasets import load_boston
+  >>> X, y = load_boston(return_X_y=True)
+  >>> from sklearn.model_selection import train_test_split
+  >>> X_train, X_test, y_train, y_test = train_test_split(X, y,
+  ...                                                     random_state=42)
+  >>> reg.fit(X_train, y_train)
+  StackingRegressor(...)
+
+During training, the `estimators` are fitted on the whole training data
+`X_train`. They will be used when calling `predict`, `predict`, or
+`predict_proba`. To generalize and avoid over-fitting, the
+`final_estimator` is trained on out-samples using internally
+:func:`sklearn.model_selection.cross_val_predict`.
+
+Note that the output of the `estimators` is controlled by the parameter
+`predict_method`. It corresponds to the method called by each
+estimator. This parameter is either a list of strings, being method names, or
```

Comment:
After reading the code and warning below, I now understand what this is doing and this description seems very unclear. also it seems pretty complicated. Is this a common use-case? Were the different usage scenarios for this discussed (sorry this is a long PR ;). I feel like giving a single string and "auto" should cover most use-cases and be easier to document and understand.
I could imagine there's someone that prefers ``decision_function`` over ``predict_proba`` and then maybe they want to change the order of priority, but I don't see why you would specify this per estimator.

Reply:
Is there any classifier in practice that we don't want to use absolutely `predict` and `predict_proba` in which it would be an issue? Otherwise, I agree with your statement. We could add this case if really this is missing. Always easier in this order.

Reply:
My point is more to remove the current per-estimator support which I find verbose and not well motivated ;)